FHEM/00_SIGNALduino.pm

```diff
@@ -2883,84 +2881,99 @@ sub SIGNALduino_IdList($@)
 	my @msIdList = ();
 	my @muIdList = ();
 	my @mcIdList = ();
-	my @skippedId = ();
+	my @skippedDevId = ();
+	my @skippedBlackId = ();
+	my @devModulId = ();
+	my %WhitelistIDs;
+	my %BlacklistIDs;
+	my $wflag = 0;		# whitelist flag, 0=disabled
+	my $bflag = 0;		# blacklist flag, 0=disabled
+	my $yflag = 0;		# 1 = alle developIDs aktivieren
+	
+	delete ($hash->{IDsNoDispatch}) if (defined($hash->{IDsNoDispatch}));
 
 	if (!defined($aVal)) {
 		$aVal = AttrVal($name,"whitelist_IDs","");
 	}
 	
-	if (!defined($blacklist)) {
-		$blacklist = AttrVal($name,"blacklist_IDs","");
-	}
-	
 	if (!defined($develop)) {
 		$develop = AttrVal($name,"development","");
 	}
-	$develop = lc($develop);
-
-	my %WhitelistIDs;
-	my %BlacklistIDs;
-	my $wflag = 0;		# whitelist flag, 0=disabled
-	my $bflag = 0;		# blacklist flag, 0=disabled
-	if (defined($aVal) && length($aVal)>0)
-	{
-		if (substr($aVal,0 ,1) eq '#') {
-			SIGNALduino_Log3 $name, 3, "$name Attr whitelist disabled: $aVal";
+	if (length($develop) > 0 && ($develop eq "1" || substr($develop,0,1) eq "y")) {	# Entwicklerversion, y ist nur zur Abwaertskompatibilitaet und kann in einer der naechsten Versionen entfernt werden
+		$yflag = 1;
+		SIGNALduino_Log3 $name, 3, "$name IdList: development version: $develop";
+	}
+	
+	if ($aVal eq "" || substr($aVal,0 ,1) eq '#') {		# whitelist nicht aktiv
+		if ($yflag == 1) {
+			SIGNALduino_Log3 $name, 3, "$name IdList attr whitelist disabled (all IDs active, except blacklist): $aVal";
 		}
 		else {
-			%WhitelistIDs = map { $_ => 1 } split(",", $aVal);
-			#my $w = join ', ' => map "$_" => keys %WhitelistIDs;
-			#SIGNALduino_Log3 $name, 3, "Attr whitelist $w";
-			$wflag = 1;
+			SIGNALduino_Log3 $name, 3, "$name IdList attr whitelist disabled (all IDs active, except blacklist and developID): $aVal";
 		}
 	}
-	if ($wflag == 0) {		# whitelist disabled
-		if (defined($blacklist) && length($blacklist)>0) {
+	else {
+		#SIGNALduino_Log3 $name, 3, "$name IdList: attr whitelistIds=$aVal";
+		%WhitelistIDs = map { $_ => 1 } split(",", $aVal);			# whitelist in Hash wandeln
+		my $w = join ', ' => map "$_" => keys %WhitelistIDs;
```

Comment:
was ist der Unterschied zu aVal? Das Leerzeichen nach dem "," ? Das hätte man auch mit search/replace in einem Schitt lösen können, wenn das wichtig ist.

Reply:
Vielleicht hast Du meine Frage übersehen, daher poste ich noch mal :)

Reply:
habe es inzwischen mit einem commit korrigiert und vergessen es hier zu erwähnen